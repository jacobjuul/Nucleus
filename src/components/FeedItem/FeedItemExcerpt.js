// @flow
import React from 'react';
import { TouchableHighlight, Text, View } from 'react-native';
import styles from './FeedItemStyles';

const FeedItemExcerpt = ({ excerpt, title }) => {
  return (
    <TouchableHighlight style={styles.content}>
      <View>
        <Text
          numberOfLines={4}
          ellipsizeMode="tail"
          style={styles.summary}
        >
          {excerpt}
        </Text>
      </View>
    </TouchableHighlight>
  );
};

export default FeedItemExcerpt;
