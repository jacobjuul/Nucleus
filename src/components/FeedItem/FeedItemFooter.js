import React                 from 'react'
import { View, Text, Image, TouchableHighlight } from 'react-native'
import styles                from './FeedItemStyles'
import pluralize             from '../../utils/pluralize'

const FeedItemFooter = ({ comments, bookmarks, onToggleBookmark, id, bookmarked }) => {
  const onBookmarkPress = () => onToggleBookmark(id, !bookmarked)
  return (
    <View style={styles.footer}>
      <View style={styles.footerSection}>
        <Image source={require('../../assets/icons/Responses.png')} />
        <Text style={[styles.footerText, styles.footerPaddingLeft]}>
          {comments.length} {pluralize('response', comments)}
        </Text>
      </View>
      <TouchableHighlight onPress={onBookmarkPress}>
        <View style={styles.footerSection}>
          <Text style={[styles.footerText, styles.footerPaddingRight]}>
            {bookmarks.length} {pluralize('bookmark', bookmarks)}
          </Text>
          {bookmarked
            ? <Image source={require('../../assets/icons/Bookmark-filled.png')} />
            : <Image source={require('../../assets/icons/bookmark-blue.png')} /> }

        </View>
      </TouchableHighlight>
    </View>
  )
}

export default FeedItemFooter
