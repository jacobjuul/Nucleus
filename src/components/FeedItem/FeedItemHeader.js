// @flow
import React                 from 'react'
import moment                from 'moment'
import { View, Image, Text } from 'react-native'
import styles                from './FeedItemStyles'

type PropTypes =
  { name: string
  , image: string
  , date: Date }

const FeedItemHeader = ({ name, image, date }: PropTypes) => {
  return (
    <View style={styles.topBar}>
      <View style={styles.topBarLeft}>
        <Image style={styles.avatar} source={{ uri: image }} />
      </View>
      <View style={styles.topBarRight}>
        <Text style={styles.author}>{name}</Text>
        <Text style={styles.date}>{moment(date).format('d. MMMM')}</Text>
      </View>
    </View>
  )
}

export default FeedItemHeader