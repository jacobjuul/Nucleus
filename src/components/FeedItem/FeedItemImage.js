import React from 'react';
import { Image } from 'react-native';

const FeedItemImage = ({ source }) =>
  <Image source={{ uri: source }} style={{ height: 200 }} resizeMode="contain" />

export default FeedItemImage;