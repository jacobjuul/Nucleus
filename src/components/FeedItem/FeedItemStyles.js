import { StyleSheet } from 'react-native'
import { colors }     from '../../constants/styles'

const FeedItemStyles = StyleSheet.create({
  container: {
    marginBottom: 15,
    paddingLeft: 15,
    paddingRight: 15,
    backgroundColor: 'white'
  },
  content: {
    paddingBottom: 16
  },
  footer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    flexDirection: 'row',
    height: 50,
    borderTopWidth: StyleSheet.hairlineWidth,
    borderColor: '#E0E5E5'
  },
  footerSection: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center'
  },
  footerText: {
    fontSize: 12,
  },
  footerPaddingLeft: { paddingLeft: 10 },
  footerPaddingRight: { paddingRight: 10 },
  title: {
    color: 'black',
    fontSize: 25,
    fontWeight: 'bold',
    marginBottom: 10,
  },
  readMore: {
    paddingTop: 15,
    paddingBottom: 15,
    color: colors.muted,
    fontSize: 12
  },
  summary: {
    fontSize: 18,
    fontFamily: 'Helvetica Neue',
    lineHeight: 28
  }
})

export default FeedItemStyles