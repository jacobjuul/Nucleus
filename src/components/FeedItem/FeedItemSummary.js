// @flow
import React from 'react';
import { TouchableHighlight, Text, View } from 'react-native';
import styles from './FeedItemStyles';

const FeedItemSummary = ({ summary, title }) => {
  return (
    <TouchableHighlight style={styles.content}>
      <View>
        <Text style={styles.title}>{title}</Text>
        <Text
          numberOfLines={4}
          ellipsizeMode="tail"
          style={styles.summary}
        >
          {summary}
        </Text>
        <TouchableHighlight>
          <Text style={styles.readMore}>Read more</Text>
        </TouchableHighlight>
      </View>
    </TouchableHighlight>
  );
};

export default FeedItemSummary;