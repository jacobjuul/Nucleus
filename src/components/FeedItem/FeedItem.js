// @flow
import React     from 'react';
import moment    from 'moment';
import {
  View,
  Text,
  StyleSheet,
  TouchableHighlight,
  Image
} from 'react-native';
import pluralize from '../../utils/pluralize';
import styles from './FeedItemStyles';
import FeedItemSummary from './FeedItemSummary';

type PropTypes =
  { title:     string
  , author:    Object
  , excerpt:   string
  , comments:  number
  , bookmarks: number };

const FeedItem = ({ title, author, excerpt, comments, bookmarks }: PropTypes) => {
  return (
    <View style={styles.container}>
      <View style={styles.topBar}>
        <View style={styles.topBarLeft}>
          <Image style={styles.avatar} source={require('../../assets/icons/logo.png')} />
        </View>
        <View style={styles.topBarRight}>
          <Text style={styles.author}>{author.email}</Text>
          <Text style={styles.date}>{moment().format('d. MMMM')}</Text>
        </View>
      </View>

      <FeedItemSummary title={title} summary={excerpt} />

      <View style={styles.footer}>
        <View><Text style={styles.footerText}>{comments} {pluralize('response', comments)}</Text></View>
        <View><Text style={styles.footerText}>{bookmarks} {pluralize('bookmark', bookmarks)}</Text></View>
      </View>
    </View>
  );
};

export default FeedItem;
