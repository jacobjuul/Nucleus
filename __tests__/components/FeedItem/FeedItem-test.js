import React from 'react'
import renderer from 'react-test-renderer'
import FeedItem from '../../../src/components/FeedItem'

describe('FeedItem', () => {
  it('renders correctly', () => {
    const author = { email: 'jacob.juul@gmail.com' }

    const instance = renderer.create(
      <FeedItem
        author={author}
        title="Short Title"
        bookmarks={4}
        comments={12}
      />
    )

    expect(instance.toJSON()).toMatchSnapshot()
  })
})
