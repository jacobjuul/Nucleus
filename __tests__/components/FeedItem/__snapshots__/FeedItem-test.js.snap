// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedItem renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "marginBottom": 15,
      "paddingLeft": 15,
      "paddingRight": 15,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 6,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "paddingBottom": 20,
        "paddingTop": 22,
      }
    }
  >
    <View
      style={Object {}}
    >
      <Image
        source={1}
        style={
          Object {
            "height": 32,
            "marginRight": 10,
            "width": 32,
          }
        }
      />
    </View>
    <View
      style={Object {}}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#69B8D6",
            "fontSize": 15,
          }
        }
      >
        jacob.juul@gmail.com
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(0,0,0,0.5)",
            "fontSize": 12,
          }
        }
      >
        5. June
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
        "fontSize": 25,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Short Title
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "paddingBottom": 16,
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={4}
        style={
          Object {
            "fontFamily": "Helvetica Neue",
            "fontSize": 18,
            "lineHeight": 28,
          }
        }
      />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgba(0,0,0,0.5)",
          "fontSize": 12,
          "paddingTop": 15,
        }
      }
    >
      Read more
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#E0E5E5",
        "borderTopWidth": 0.5,
        "display": "flex",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-between",
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        12
         
        responses
      </Text>
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        4
         
        bookmarks
      </Text>
    </View>
  </View>
</View>
`;
